src/funcx_common/task_storage/chain.py

```diff
@@ -0,0 +1,55 @@
+import typing as t
+
+from ..tasks import TaskProtocol
+from .base import StorageException, TaskStorage
+
+
+class ChainedTaskStorage(TaskStorage):
+    """
+    Chained storage combines multiple other concrete storage options.
+
+    The storage methods are used on a "first-come, first-served" basis, with the
+    intended usage pattern being
+
+    >>> slow_storage = ...
+    >>> fast_storage = ...
+    >>> task_storage = ChainedTaskStorage(fast_storage, slow_storage)
+
+    In this scenario, the fast storage will be tried first for both the store and get
+    operations.
+
+    If the fast storage rejects a storage operation, the slow storage will be tried. A
+    failure (exception) would be raised immediately, however.
+
+    The assumption here is that the fast storage responds quickly enough that it can be
+    checked first in all cases, even without additional context for the call.
+    """
+
+    storage_id = "ChainedTaskStorage"
+
+    def __init__(self, *storages: TaskStorage) -> None:
+        self._storages = list(storages)
+        self.storage_map = {storage.storage_id: storage for storage in storages}
```

Comment:
This is mixing `self._storages` (indicating a private field) and `self.storage_map` (not indicating a private field).

It doesn't strike me as intentional. I think either both should have a leading underscore or both should not, unless there's a reason for only one of the two to be considered a public part of the class interface.

The fact that they are mixed is not only inconsistent, but it could be misread as a signal that `storage_map` is public and `_storages` is not -- that's the worst-case scenario, since updating one without updating the other could cause bugs.

Reply:
Fixed.